src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,189 @@
+package duke;
+
+import duke.exception.*;
+
+import java.util.ArrayList;
+
+/**
+ * A class that represents a list of Tasks.
+ *
+ * It contains a constructor, a getter,
+ * a method to print a list, add a task to a list,
+ * mark a task as done and delete a task.
+ *
+ */
+public class TaskList {
+    private ArrayList<Task> list;
```

Comment:
Perhaps it would be better to name this variable tasks? Since it is a collection of Task objects.

```suggestion
    private ArrayList<Task> tasks;
```